app/script/util/util.js

```diff
@@ -779,6 +779,54 @@ z.util.afterRender = callback => {
 
 /**
  * No operation
- * @returns {undefined}
+ * @returns {void}
  */
 z.util.noop = function() {};
+
+/*
+ * Call Logging function
+ */
+z.util.CallingLog = [];
+z.util.CallingLogger = class CallingLogger extends z.util.Logger {
+  constructor(message, options) {
+    super(message, options);
+  }
+
+  static get CONFIG() {
+    return {
+      MESSAGE_LOG_LENGTH: 10000,
+    };
+  }
+
+  _get_debug_type(number) {
+    switch (number) {
+      case 300:
+      case 400:
+      case 500:
+      case 700:
+        return 'DEBUG';
+      case 800:
+        return 'INFO';
+      case 900:
+        return 'WARNING';
+      case 100:
+        return 'ERROR';
+    }
+
+    return '?';
+  }
+
+  _log_to_memory(args) {
+    while (z.util.CallingLog.length >= CallingLogger.CONFIG.MESSAGE_LOG_LENGTH) {
+      z.util.CallingLog.shift();
+    }
+
+    const logMessage = `[${new Date().toISOString()}] (${this._get_debug_type(args[0]())}) ${args[1]}`;
+    z.util.CallingLog.push(logMessage);
+  }
+
+  _print_log(args) {
```

Comment:
everywhere

Reply:
`_print_log` is from the logging library and cannot be changed